### PR TITLE
[REFACTOR] CDPMessageBus not static anymore; fixes #1310

### LINKED ISSUE #1310: [REFACTOR] CDPMessageBus not static anymore
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-IME
- [x] I have searched [open](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/issues) and [closed](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Since CDP4-COMET SDK 25.0.0 the CDPMessageBus is not a static classs anymore.
MessageBus should be made available via the ICDPMessageBus interface using dependency injection.

### PR DESCRIPTION
**Pull request by lxatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
[WIP] CDPMessageBus not static anymore

### CODE REVIEW

File: BasicRdl/ViewModels/Ribbons/SiteRdlSelectionRibbonViewModel.cs [hunk 1]
```diff
@@ -67,18 +62,24 @@ public class SiteRdlSelectionRibbonViewModel : ReactiveObject
...
+        /// </param>
+        public SiteRdlSelectionRibbonViewModel(IDialogNavigationService dialogNavigationService, ICDPMessageBus messageBus)
         {
-            this.thingDialogNavigationService = ServiceLocator.Current.GetInstance<IThingDialogNavigationService>();
+            this.dialogNavigationService = dialogNavigationService;
 
             this.OpenSessions = new ReactiveList<ISession>();
             this.OpenSessions.CountChanged.Select(x => x != 0).ToProperty(this, x => x.HasSession, out this.hasSession, scheduler: RxApp.MainThreadScheduler);
-            CDPMessageBus.Current.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
+            messageBus.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
```
Comment (antoineatrhea) [inline BasicRdl/ViewModels/Ribbons/SiteRdlSelectionRibbonViewModel.cs:82]: this subscription should be disposed, no?

File: CDP4Addin/Addin.cs [hunk 2]
```diff
@@ -421,17 +428,17 @@ private void SetupIdtExtensibility2Events()
         /// </summary>
         private void SetupEventListeners()
```
Comment (antoineatrhea) [inline CDP4Addin/Addin.cs:429]: Should we not dispose those subscriptions?

File: CDP4Addin/OfficeRibbon/AddinRibbonPart.cs [hunk 3]
```diff
@@ -85,10 +85,13 @@ public class AddinRibbonPart : RibbonPart
...
-            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService)
+        /// <param name="messageBus">
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        public AddinRibbonPart(int order, IPanelNavigationService panelNavigationService, IThingDialogNavigationService thingDialogNavigationService, IDialogNavigationService dialogNavigationService, IPluginSettingsService pluginSettingsService, IAppSettingsService<AddinAppSettings> appSettingService, ICDPMessageBus messageBus)
+            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService, messageBus)
         {
-            CDPMessageBus.Current.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
+            messageBus.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
```
Comment (antoineatrhea) [inline CDP4Addin/OfficeRibbon/AddinRibbonPart.cs:94]: Same here

File: CDP4Composition/Mvvm/MenuItems/RibbonButtonEngineeringModelSetupDependentViewModel.cs [hunk 4]
```diff
@@ -63,23 +63,26 @@ public abstract class RibbonButtonEngineeringModelSetupDependentViewModel : Reac
         /// <param name="instantiatePanelViewModelFunction">
         /// The instantiate Panel View Model Function.
         /// </param>
-        protected RibbonButtonEngineeringModelSetupDependentViewModel(Func<EngineeringModelSetup, ISession, IThingDialogNavigationService, IPanelNavigationService, IDialogNavigationService, IPluginSettingsService, IPanelViewModel> instantiatePanelViewModelFunction)
+        /// <param name="messageBus">
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        protected RibbonButtonEngineeringModelSetupDependentViewModel(Func<EngineeringModelSetup, ISession, IThingDialogNavigationService, IPanelNavigationService, IDialogNavigationService, IPluginSettingsService, IPanelViewModel> instantiatePanelViewModelFunction, ICDPMessageBus messageBus)
```
Comment (antoineatrhea) [inline CDP4Composition/Mvvm/MenuItems/RibbonButtonEngineeringModelSetupDependentViewModel.cs:69]: Same here

File: CDP4Composition/Mvvm/MenuItems/RibbonButtonIterationDependentViewModel.cs [hunk 5]
```diff
@@ -64,7 +64,10 @@ public class RibbonButtonIterationDependentViewModel : ReactiveObject
         /// <param name="instantiatePanelViewModelFunction">
         /// The instantiate Panel View Model Function.
         /// </param>
-        protected RibbonButtonIterationDependentViewModel(Func<Iteration, ISession, IThingDialogNavigationService, IPanelNavigationService, IDialogNavigationService, IPluginSettingsService, IPanelViewModel> instantiatePanelViewModelFunction)
+        /// <param name="messageBus">
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        protected RibbonButtonIterationDependentViewModel(Func<Iteration, ISession, IThingDialogNavigationService, IPanelNavigationService, IDialogNavigationService, IPluginSettingsService, IPanelViewModel> instantiatePanelViewModelFunction, ICDPMessageBus messageBus)
```
Comment (antoineatrhea) [inline CDP4Composition/Mvvm/MenuItems/RibbonButtonIterationDependentViewModel.cs:70]: Same

File: CDP4Composition/Mvvm/MenuItems/RibbonButtonOptionDependentViewModel.cs [hunk 6]
```diff
@@ -63,38 +63,38 @@ public abstract class RibbonButtonOptionDependentViewModel : ReactiveObject
         /// <param name="instantiatePanelViewModelFunction">
         /// The instantiate Panel View Model Function.
         /// </param>
-        protected RibbonButtonOptionDependentViewModel(Func<Option, ISession, IThingDialogNavigationService, IPanelNavigationService, IDialogNavigationService, IPluginSettingsService, IPanelViewModel> instantiatePanelViewModelFunction)
+        protected RibbonButtonOptionDependentViewModel(Func<Option, ISession, IThingDialogNavigationService, IPanelNavigationService, IDialogNavigationService, IPluginSettingsService, IPanelViewModel> instantiatePanelViewModelFunction, ICDPMessageBus messageBus)
```
Comment (antoineatrhea) [inline CDP4Composition/Mvvm/MenuItems/RibbonButtonOptionDependentViewModel.cs:66]: Same

File: CDP4Composition/Mvvm/MenuItems/RibbonButtonSessionDependentViewModel.cs [hunk 7]
```diff
@@ -66,7 +66,7 @@ protected RibbonButtonSessionDependentViewModel(Func<ISession, IThingDialogNavig
 
             this.OpenSingleBrowserCommand = ReactiveCommandCreator.Create(this.ExecuteOpenSingleBrowser);
 
-            CDPMessageBus.Current.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
+            messageBus.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
```
Comment (antoineatrhea) [inline CDP4Composition/Mvvm/MenuItems/RibbonButtonSessionDependentViewModel.cs:69]: Same

File: CDP4Composition/Mvvm/MenuItems/RibbonButtonSessionFavoritesDependentViewModel.cs [hunk 8]
```diff
@@ -59,7 +59,7 @@ public abstract class RibbonButtonSessionFavoritesDependentViewModel : ReactiveO
         /// <summary>
         /// Initializes a new instance of the <see cref="RibbonButtonSessionFavoritesDependentViewModel"/> class
         /// </summary>
-        protected RibbonButtonSessionFavoritesDependentViewModel(Func<ISession, IThingDialogNavigationService, IPanelNavigationService, IDialogNavigationService, IPluginSettingsService, IFavoritesService, IPanelViewModel> instantiatePanelViewModel)
+        protected RibbonButtonSessionFavoritesDependentViewModel(Func<ISession, IThingDialogNavigationService, IPanelNavigationService, IDialogNavigationService, IPluginSettingsService, IFavoritesService, IPanelViewModel> instantiatePanelViewModel, ICDPMessageBus messageBus)
```
Comment (antoineatrhea) [inline CDP4Composition/Mvvm/MenuItems/RibbonButtonSessionFavoritesDependentViewModel.cs:62]: Same

File: CDP4Composition/Views/CustomFilterEditorDialog.xaml.cs [hunk 9]
```diff
@@ -60,7 +60,9 @@ public CustomFilterEditorDialog(bool initializeComponent)
                 this.InitializeComponent();
             }
 
-            CDPMessageBus.Current.Listen<ApplyFilterEvent>()
+            var messageBus = CommonServiceLocator.ServiceLocator.Current.GetInstance<ICDPMessageBus>();
+
+            messageBus.Listen<ApplyFilterEvent>()
```
Comment (antoineatrhea) [inline CDP4Composition/Views/CustomFilterEditorDialog.xaml.cs:65]: Same

File: CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs [hunk 10]
```diff
@@ -104,16 +104,19 @@ public class CrossViewEditorRibbonPart : RibbonPart
...
+        /// </param>
+        public CrossViewEditorRibbonPart(int order, IPanelNavigationService panelNavigationService, IThingDialogNavigationService thingDialogNavigationService, IDialogNavigationService dialogNavigationService, IPluginSettingsService pluginSettingsService, IOfficeApplicationWrapper officeApplicationWrapper, ICDPMessageBus messageBus)
+            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService, messageBus)
         {
             this.officeApplicationWrapper = officeApplicationWrapper;
             this.ExcelQuery = new ExcelQuery();
             this.Iterations = new List<Iteration>();
 
-            CDPMessageBus.Current.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
+            messageBus.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
```
Comment (antoineatrhea) [inline CDP4CrossViewEditor/OfficeRibbon/CrossViewEditorRibbonPart.cs:117]: Same

File: CDP4LogInfo/ViewModels/Controls/LogInfoControlsViewModel.cs [hunk 11]
```diff
@@ -50,13 +55,19 @@ public class LogInfoControlsViewModel : ReactiveObject
...
+        /// </param>
+        public LogInfoControlsViewModel(IDialogNavigationService dialogNavigationService, ICDPMessageBus messageBus)
         {
             this.LogInfoPanel = new LogInfoPanelViewModel(dialogNavigationService);
 
             this.OpenClosePanelCommand = ReactiveCommandCreator.Create(this.ExecuteOpenClosePanel);
-            
-            CDPMessageBus.Current.Listen<NavigationPanelEvent>()
+
+            messageBus.Listen<NavigationPanelEvent>()
```
Comment (antoineatrhea) [inline CDP4LogInfo/ViewModels/Controls/LogInfoControlsViewModel.cs:70]: same

File: CDP4ParameterSheetGenerator/OfficeRibbon/ParameterSheetGeneratorRibbonPart.cs [hunk 12]
```diff
@@ -68,17 +95,20 @@ public class ParameterSheetGeneratorRibbonPart : RibbonPart
...
-        /// </param>        
-        public ParameterSheetGeneratorRibbonPart(int order, IPanelNavigationService panelNavigationService, IThingDialogNavigationService thingDialogNavigationService, IDialogNavigationService dialogNavigationService, IPluginSettingsService pluginSettingsService, IOfficeApplicationWrapper officeApplicationWrapper)
-            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService)
+        /// </param>
+        /// <param name="messageBus">
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        public ParameterSheetGeneratorRibbonPart(int order, IPanelNavigationService panelNavigationService, IThingDialogNavigationService thingDialogNavigationService, IDialogNavigationService dialogNavigationService, IPluginSettingsService pluginSettingsService, IOfficeApplicationWrapper officeApplicationWrapper, ICDPMessageBus messageBus)
+            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService, messageBus)
```
Comment (antoineatrhea) [inline CDP4ParameterSheetGenerator/OfficeRibbon/ParameterSheetGeneratorRibbonPart.cs:103]: same

File: CDP4PropertyGrid/ViewModels/ViewRibbonControlViewModel.cs [hunk 13]
```diff
@@ -52,11 +57,15 @@ public class ViewRibbonControlViewModel : ReactiveObject
...
+        /// <param name="messageBus">
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        public ViewRibbonControlViewModel(ICDPMessageBus messageBus)
         {
+            this.CDPMessageBus = messageBus;
             this.OpenClosePanelCommand = ReactiveCommandCreator.Create(this.ExecuteOpenClosePanel);
 
-            CDPMessageBus.Current.Listen<NavigationPanelEvent>()
+            this.CDPMessageBus.Listen<NavigationPanelEvent>()
```
Comment (antoineatrhea) [inline CDP4PropertyGrid/ViewModels/ViewRibbonControlViewModel.cs:68]: same

File: CDP4Scripting/ViewModels/ScriptingEngineRibbonPageGroupViewModel.cs [hunk 14]
```diff
@@ -84,28 +85,46 @@ public class ScriptingEngineRibbonPageGroupViewModel : ReactiveObject
...
+        /// <param name="fileDialogService">
+        /// The <see cref="IOpenSaveFileDialogService"/>
+        /// </param>
+        /// <param name="scriptingProxy">
+        /// The <see cref="IScriptingProxy"/>
+        /// </param>
+        /// <param name="messageBus">
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        public ScriptingEngineRibbonPageGroupViewModel(IPanelNavigationService panelNavigationService, IOpenSaveFileDialogService fileDialogService, IScriptingProxy scriptingProxy, ICDPMessageBus messageBus)
```
Comment (antoineatrhea) [inline CDP4Scripting/ViewModels/ScriptingEngineRibbonPageGroupViewModel.cs:108]: Same

File: CDP4SiteDirectory/OfficeRibbon/SiteDirectoryRibbonPart.cs [hunk 15]
```diff
@@ -106,12 +106,15 @@ public class SiteDirectoryRibbonPart : RibbonPart
...
-        public SiteDirectoryRibbonPart(int order, IPanelNavigationService panelNavigationService, IThingDialogNavigationService thingDialogNavigationService, IDialogNavigationService dialogNavigationService, IPluginSettingsService pluginSettingsService)
-            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService)
+        /// <param name="messageBus">
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        public SiteDirectoryRibbonPart(int order, IPanelNavigationService panelNavigationService, IThingDialogNavigationService thingDialogNavigationService, IDialogNavigationService dialogNavigationService, IPluginSettingsService pluginSettingsService, ICDPMessageBus messageBus)
+            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService, messageBus)
         {
-            CDPMessageBus.Current.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
+            messageBus.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
```
Comment (antoineatrhea) [inline CDP4SiteDirectory/OfficeRibbon/SiteDirectoryRibbonPart.cs:115]: same

File: CDP4SiteDirectory/ViewModels/HighlightingRibbonViewModel.cs [hunk 16]
```diff
@@ -56,23 +61,24 @@ public class HighlightingRibbonViewModel : ReactiveObject
...
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        public HighlightingRibbonViewModel(ICDPMessageBus messageBus)
         {
+            this.messageBus = messageBus;
             this.openSessions = new ReactiveList<ISession>();
             this.openSessions.CountChanged.Select(x => x != 0).ToProperty(this, x => x.HasSession, out this.hasSession, scheduler: RxApp.MainThreadScheduler);
 
-            CDPMessageBus.Current.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
+            this.messageBus.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
```
Comment (antoineatrhea) [inline CDP4SiteDirectory/ViewModels/HighlightingRibbonViewModel.cs:73]: same

File: CDP4SiteDirectory/ViewModels/ShowDeprecatedRibbonViewModel.cs [hunk 17]
```diff
@@ -69,14 +74,18 @@ public class ShowDeprecatedBrowserRibbonViewModel : ReactiveObject, IDeprecatabl
...
         {
+            this.messageBus = messageBus;
             this.filterStringService = ServiceLocator.Current.GetInstance<IFilterStringService>();
-            
+
             this.openSessions = new ReactiveList<ISession>();
             this.openSessions.CountChanged.Select(x => x != 0).ToProperty(this, x => x.HasSession, out this.hasSession, scheduler: RxApp.MainThreadScheduler);
 
-            CDPMessageBus.Current.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
+            this.messageBus.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
```
Comment (antoineatrhea) [inline CDP4SiteDirectory/ViewModels/ShowDeprecatedRibbonViewModel.cs:88]: same

File: EngineeringModel/OfficeRibbon/EngineeringModelRibbonPart.cs [hunk 18]
```diff
@@ -124,8 +123,8 @@ public class EngineeringModelRibbonPart : RibbonPart
...
-            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService)
+        public EngineeringModelRibbonPart(int order, IPanelNavigationService panelNavigationService, IDialogNavigationService dialogNavigationService, IThingDialogNavigationService thingDialogNavigationService, IPluginSettingsService pluginSettingsService, IParameterSubscriptionBatchService parameterSubscriptionBatchService, IParameterActualFiniteStateListApplicationBatchService parameterActualFiniteStateListApplicationBatchService, IChangeOwnershipBatchService changeOwnershipBatchService, ICDPMessageBus messageBus)
+            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService, messageBus)
```
Comment (antoineatrhea) [inline EngineeringModel/OfficeRibbon/EngineeringModelRibbonPart.cs:127]: Same

File: ProductTree/OfficeRibbon/ProductTreeRibbonPart.cs [hunk 19]
```diff
@@ -84,19 +84,22 @@ public class ProductTreeRibbonPart : RibbonPart
         /// <param name="pluginSettingsService">
         /// The <see cref="IPluginSettingsService"/> used to read and write plugin setting files.
         /// </param>
-        public ProductTreeRibbonPart(int order, IPanelNavigationService panelNavigationService, IDialogNavigationService dialogNavigationService, IThingDialogNavigationService thingDialogNavigationService, IPluginSettingsService pluginSettingsService)
-            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService)
+        /// <param name="messageBus">
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        public ProductTreeRibbonPart(int order, IPanelNavigationService panelNavigationService, IDialogNavigationService dialogNavigationService, IThingDialogNavigationService thingDialogNavigationService, IPluginSettingsService pluginSettingsService, ICDPMessageBus messageBus)
+            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService, messageBus)
```
Comment (antoineatrhea) [inline ProductTree/OfficeRibbon/ProductTreeRibbonPart.cs:91]: Same

File: Requirements/OfficeRibbon/RequirementRibbonPart.cs [hunk 20]
```diff
@@ -78,17 +103,21 @@ public class RequirementRibbonPart : RibbonPart
         /// <param name="officeApplicationWrapper">
         /// The instance of <see cref="IOfficeApplicationWrapper"/> that provides access to the loaded Office application.
         /// </param>
-        public RequirementRibbonPart(int order, IPanelNavigationService panelNavigationService, IDialogNavigationService dialogNavigationService, IThingDialogNavigationService thingDialogNavigationService, IPluginSettingsService pluginSettingsService, IOfficeApplicationWrapper officeApplicationWrapper)
-            : base(order, panelNavigationService, thingDialogNavigationService, dialogNavigationService, pluginSettingsService)
+        /// <param name="messageBus">
+        /// The <see cref="ICDPMessageBus"/>
+        /// </param>
+        public RequirementRibbonPart(int order, IPanelNavigationService panelNavigationService, IDialogNavigationService dialogNavigationService, IThingDialogNavigationService thingDialogNavigationService, IPluginSettingsService pluginSettingsService, IOfficeApplicationWrapper officeApplicationWrapper, ICDPMessageBus messageBus)
```
Comment (antoineatrhea) [inline Requirements/OfficeRibbon/RequirementRibbonPart.cs:109]: same

File: Requirements/ViewModels/HtmlReport/HtmlExportRibbonViewModel.cs [hunk 21]
```diff
@@ -69,18 +74,15 @@ public HtmlExportRibbonViewModel()
...
             this.dialogNavigationService = ServiceLocator.Current.GetInstance<IDialogNavigationService>();
             this.openSaveFileDialogService = ServiceLocator.Current.GetInstance<IOpenSaveFileDialogService>();
+            this.CDPMessageBus = ServiceLocator.Current.GetInstance<ICDPMessageBus>();
 
             this.Sessions = new ReactiveList<ISession>();
             this.Iterations = new ReactiveList<Iteration>();
 
-            CDPMessageBus.Current.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
-            CDPMessageBus.Current.Listen<ObjectChangedEvent>(typeof(Iteration)).Subscribe(this.IterationEventHandler);
+            this.CDPMessageBus.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
```
Comment (antoineatrhea) [inline Requirements/ViewModels/HtmlReport/HtmlExportRibbonViewModel.cs:82]: same

File: Requirements/ViewModels/ReqIF/ReqIfRibbonViewModel.cs [hunk 22]
```diff
@@ -87,16 +92,17 @@ public ReqIfRibbonViewModel()
             this.ThingDialogNavigationService = ServiceLocator.Current.GetInstance<IThingDialogNavigationService>();
             this.PluginSettingsService = ServiceLocator.Current.GetInstance<IPluginSettingsService>();
             this.OpenSaveDialogService = ServiceLocator.Current.GetInstance<IOpenSaveFileDialogService>();
+            this.CDPMessageBus = ServiceLocator.Current.GetInstance<ICDPMessageBus>();
 
             this.Sessions = new ReactiveList<ISession>();
             this.Iterations = new ReactiveList<Iteration>();
 
-            CDPMessageBus.Current.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
+            this.CDPMessageBus.Listen<SessionEvent>().Subscribe(this.SessionChangeEventHandler);
```
Comment (antoineatrhea) [inline Requirements/ViewModels/ReqIF/ReqIfRibbonViewModel.cs:100]: same

**Review by antoineatrhea — APPROVED** [review]
Dunno if those non-disposed subscriptions have any impact and if they have to be disposed (automatically cleaned when the application closes?)

**Comment by lxatrhea** [comment]
> Dunno if those non-disposed subscriptions have any impact and if they have to be disposed (automatically cleaned when the application closes?)

Ribbon ViewModels only close at closure of the app. No need to Dispose it as that will happen automatically.

File: CDP4Composition/Views/CustomFilterEditorDialog.xaml.cs (hunk 9 above)
Comment (lxatrhea) [reply CDP4Composition/Views/CustomFilterEditorDialog.xaml.cs]: Updated in new commit

File: CDP4ObjectBrowser/AutoGen/ReferenceDataLibraryRowViewModel.cs [hunk 23]
```diff
@@ -106,7 +106,7 @@ protected ReferenceDataLibraryRowViewModel(T referenceDataLibrary, ISession sess
             this.constantFolder = new CDP4Composition.FolderRowViewModel("Constant", "Constant", this.Session, this);
             this.ContainedRows.Add(this.constantFolder);
 
-            var rdlListener = this.messageBus.Listen<SessionEvent>()
+            var rdlListener = this.CDPMessageBus.Listen<SessionEvent>()
```
Comment (samatrhea) [inline CDP4ObjectBrowser/AutoGen/ReferenceDataLibraryRowViewModel.cs:109]: this is generated, did you update the generator?

**Review by samatrhea — CHANGES_REQUESTED** [review]

File: CDP4ObjectBrowser/AutoGen/ReferenceDataLibraryRowViewModel.cs (hunk 23 above)
Comment (lxatrhea) [reply CDP4ObjectBrowser/AutoGen/ReferenceDataLibraryRowViewModel.cs:109]: Issue created to refactor T4 scripts: #1309

**Pull request merged**